ethcore/src/evm/wasm/mod.rs

```diff
@@ -151,49 +151,106 @@ impl evm::Evm for WasmInterpreter {
 }
 
 fn native_bindings<'a>(runtime: &'a mut Runtime) -> interpreter::UserFunctions<'a> {
+	use parity_wasm::elements::ValueType::*;
+
 	interpreter::UserFunctions {
 		executor: runtime,
 		functions: vec![
 			interpreter::UserFunction {
 				name: "_storage_read".to_owned(),
-				params: vec![elements::ValueType::I32, elements::ValueType::I32],
-				result: Some(elements::ValueType::I32),
+				params: vec![I32, I32],
```

Comment:
since these all seem to be statically known, why not have it take a `&'static [ValueType]` instead of requiring allocation on each one?